### ISSUE TITLE
Nested class with no snapshots results in java.nio.file.NoSuchFileException

### ISSUE THREAD

**Issue by GuyJackson** [origin-energy/java-snapshot-testing#59]
```
@ExtendWith({SnapshotExtension.class})
class SnapshotDemoApplicationTest {

  @Nested
  class NestedWithSnapshotClass {

    @Test
    void nested_normal_test() {
      assertThat(true).isTrue();
    }

    @Test
    void nested_snapshot_test() {
      SnapshotMatcher.expect("nested hello").toMatchSnapshot();
    }
  }

  @Nested
  class NestedWithNoSnapShots {
    
    @Test
    void nested_normal_test() {
      assertThat(true).isTrue();
    }
  }
}
```
snapshot version: **2.2.2**
junit-jupiter version: **5.7.0**

The tests within `NestedWithSnapshotClass` run as expected.

The tests within `NestedWithNoSnapShots` has an error `java.nio.file.NoSuchFileException` - though the actual test seems to pass. Output:

![image](https://user-images.githubusercontent.com/5394347/103901091-99583500-50f0-11eb-9759-cb2ec6f1db67.png)

**Comment by jackmatt2** [comment]
This bug is related to the introduction of on demand snapshot file creation.  We should probably just catch this exception and continue in this case.

**Comment by jackmatt2** [comment]
Fixed as of release 2.3.0

Note: you need to manually add these dependencies now too

```
    // Required java-snapshot-testing peer dependencies
   testImplementation 'com.fasterxml.jackson.core:jackson-core:2.11.3'
   testImplementation 'com.fasterxml.jackson.core:jackson-databind:2.11.3'
   testImplementation 'com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.11.3'
   testImplementation 'com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.11.3'
```
